### ISSUE TITLE
Crutches.Format.Number.as_rounded

### ISSUE THREAD

**Issue by mykewould** [mykewould/crutches#29]
```
@doc ~S"""
Formats a `number` with the specified level of :precision (e.g., 112.32 has a 
precision of 2 if `:significant` is false, and 5 if `:significant` is true). You
can customize the format in the `options` Dict.

# Options

  * `:locale` - Sets the locale to be used for formatting (defaults to current locale).
  * `:precision` - Sets the precision of the number (defaults to 3).
  * `:significant` - If true, precision will be the # of significant_digits. If false, the # of fractional digits (defaults to false).
  * `:separator` - Sets the separator between the fractional and integer digits (defaults to “.”).
  * `:delimiter` - Sets the thousands delimiter (defaults to “”).
  * `:strip_insignificant_zeros` - If true removes insignificant zeros after the decimal separator (defaults to false).

# Examples

    iex> Number.as_rounded(111.2345)                                  
    "111.235"

    iex> Number.as_rounded(111.2345, precision: 2)                    
    "111.23"

    iex> Number.as_rounded(13, precision: 5)                          
    "13.00000"

    iex> Number.as_rounded(389.32314, precision: 0)                   
    "389"

    iex> Number.as_rounded(111.2345, significant: true)               
    "111"

    iex> Number.as_rounded(111.2345, precision: 1, significant: true) 
    "100"

    iex> Number.as_rounded(13, precision: 5, significant: true)       
    "13.000"

    iex> Number.as_rounded(111.234, locale: :fr)                      
    "111,234"

    iex> Number.as_rounded(13, precision: 5, significant: true, strip_insignificant_zeros: true)
    "13"

    iex> Number.as_rounded(389.32314, precision: 4, significant: true) 
    "389.3"

    iex> Number.as_rounded(1111.2345, precision: 2, separator: ',', delimiter: '.') 
    "1.111,23"

"""
```

**Comment by druzn3k** [comment]
updated description

**Comment by CyrusNuevoDia** [comment]
Taking this on right now :shipit:

**Comment by duijf** [comment]
Closed with the merge of #64 in 8c3baf4.
